provisioning/device/src/transport_state_machine.ts

```diff
@@ -0,0 +1,239 @@
+        },
+        sendingRegistrationReqest: {
+          _onEnter: (callback, registrationId, authorization, requestBody, forceRegistration) => {
+            /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_012: [ `register` shall call `TransportHandlers.registrationRequest`. ] */
+            this._transport.registrationRequest(registrationId, authorization, requestBody, forceRegistration, (err, responseBody, result, pollingInterval) => {
+              this._fsm.handle('responseReceived', callback, err, registrationId, responseBody, result, pollingInterval);
+            });
+          },
+          responseReceived: (callback, err, registrationId, responseBody, result, pollingInterval) => {
+            this._fsm.transition('analyzingResponse', callback, err, registrationId, responseBody, result, pollingInterval);
+          },
+          disconnect: (callback) => this._fsm.transition('disconnecting', callback),
+          /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_006: [ If `connect` is called while the transport is executing the first registration request, it shall do nothing and call `callback` immediately. ] */
+          connect: (callback) => callback(),
+          /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_024: [ If `register` is called while a different request is in progress, it shall fail with an `InvalidOperationError`. ] */
+          register: (callback) => callback(new errors.InvalidOperationError('another operation is in progress'))
+        },
+        analyzingResponse: {
+          _onEnter: (callback, err, registrationId, responseBody, result, pollingInterval) => {
+            if (err) {
+              /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_013: [ If `TransportHandlers.registrationRequest` fails, `register` shall fail. ] */
+              /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_019: [ If `TransportHandlers.queryOperationStatus` fails, `register` shall fail. ] */
+              this._fsm.transition('connected', callback, err);
+            } else if (responseBody) {
+              debug('received response from service:' + JSON.stringify(responseBody));
+              switch (responseBody.status.toLowerCase()) {
+                case 'assigned': {
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_014: [ If `TransportHandlers.registrationRequest` succeeds with status==Assigned, it shall emit an 'operationStatus' event and call `callback` with null, the response body, and the protocol-specific result. ] */
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_020: [ If `TransportHandlers.queryOperationStatus` succeeds with status==Assigned, `register` shall emit an 'operationStatus' event and complete and pass the body of the response and the protocol-spefic result to the `callback`. ] */
+                  this.emit('operationStatus', responseBody);
```

Comment:
>this.emit('operationStatus', responseBody); [](start = 18, length = 43)

emit is synchronous meaning you're going to get stuck in that state until the customer-defined callback finishes. I think you're better off transitioning back into connected (or a "registrationComplete" state, not sure), and emitting from there.